File: benchmark/output/python-pygame/gemini/simple_animation.py
import pygame
import sys

pygame.init()
screen = pygame.display.set_mode((800, 600))
pygame.display.set_caption("Pulsing Rectangle")
clock = pygame.time.Clock()
font = pygame.font.Font(None, 36)

# Game state variables
frame = 0
frame_timer = 0.0

running = True
while running:
    dt = clock.tick(60) / 1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False

    # Update logic
    frame_timer += dt
    if frame_timer >= 0.25:
        frame_timer -= 0.25
        frame += 1
        if frame >= 4:
            frame = 0

    # Draw logic
    screen.fill((0, 0, 0))
    
    if frame == 0:
        rect = (380.0, 280.0, 40.0, 40.0)
    elif frame == 1:
        rect = (375.0, 275.0, 50.0, 50.0)
    elif frame == 2:
        rect = (370.0, 270.0, 60.0, 60.0)
    else: # frame == 3
        rect = (375.0, 275.0, 50.0, 50.0)
        
    pygame.draw.rect(screen, (255, 255, 255), rect)
    
    text = font.render("Frame: " + str(frame), True, (255, 255, 255))
    screen.blit(text, (10.0, 10.0))

    pygame.display.flip()

pygame.quit()
sys.exit()

File: vibe-lang/benchmark/output/lua-love/gemini/simple_animation.lua
local frame = 0
local frame_timer = 0.0

function love.load()
    love.window.setMode(800, 600)
    love.window.setTitle("Pulsing Rectangle")
end

function love.update(dt)
    frame_timer = frame_timer + dt
    
    if frame_timer >= 0.25 then
        frame_timer = frame_timer - 0.25
        frame = frame + 1
    end
    
    if frame >= 4 then
        frame = 0
    end
end

function love.draw()
    if frame == 0 then
        love.graphics.rectangle("fill", 380, 280, 40, 40)
    elseif frame == 1 then
        love.graphics.rectangle("fill", 375, 275, 50, 50)
    elseif frame == 2 then
        love.graphics.rectangle("fill", 370, 270, 60, 60)
    elseif frame == 3 then
        love.graphics.rectangle("fill", 375, 275, 50, 50)
    end
    
    love.graphics.print("Frame: " .. tostring(frame), 10, 10)
end